### DISCUSSION TITLE
State management problems

### DISCUSSION THREAD

**Discussion by kabaskill** [neuefische/questions#804] [Web] [answered]
Hello,
I am currently working on state management for our `FavoriteButton` component. The states persists through the whole app, if you are using the links on `navBar`. But clicking on the `back` button on the details page causes a re-render, resetting the state. That brings me to my question.

Why the re-render happens, even though the `navBar` uses the same `Link` component with the `Back` button?

The current state of the code can be found on the `searchPage` branch of the repository below.
[GameDalf GitHub Repository](https://github.com/itsmyrice/gamedalf/tree/searchPage)

I'm pretty sure this issue is solvable with `useLocalStorageState` or maybe `useMemo` even, but I'm curious about why the re-render happens anyway.

Thank you and have a great day.

**Comment by onemanwenttomow** [comment, marked as answer]
I think the issue is that your back button is a `StyledLink`. Which is and `a` tag rather than the `Link` component from next.js

**Reply by kabaskill** [reply]
Wow! That's one sneaky misfit of a code. I changed it to a `Link` component  from an `a` tag and it works as intended.
Thanks!
